packages/marshal/src/capDataJSON.js

```diff
@@ -0,0 +1,38 @@
+// @ts-check
+
+const { Fail } = assert;
+
+/** @param {unknown} x */
+const assertJSON = x => {
+  assert.typeof(x, 'string');
+  void JSON.parse(x);
+};
+
+/**
+ * @param {import('./types').CapData<unknown>} capData - with "simple" slots;
+ *   that is: slots whose JSON form has no occurrence of `:[`.
+ * @returns {string}
+ */
+export const capDataToJSON = ({ body, slots }) => {
+  assert(Array.isArray(slots));
+  const slotj = JSON.stringify(slots);
+  slotj.indexOf(':[') < 0 || Fail`expected simple slots`;
+  const body1 = body.replace(/^#/, '');
+  assertJSON(body1);
+  const json = `{"$body":${body1},"slots":${slotj}}`;
+  assertJSON(json);
+  return json;
+};
+
+export const JSONToCapData = json => {
+  assert.typeof(json, 'string');
+  json.startsWith('{"$body":') || Fail`expected $body`;
```

Comment:
I guess this only works when this body is first in the serialized JSON, not second?